src/main/java/seedu/edulog/model/calendar/Day.java

```diff
@@ -0,0 +1,126 @@
+    public static final List<String> DAYS_OF_THE_WEEK_SHORTHAND =
+        DAYS_OF_THE_WEEK.stream().map(s -> s.substring(0, 3)).collect(Collectors.toList());
+
+    public static final String INVALID_DAY_OF_WEEK =
+        "Day of the week must be spelt as "
+            + "'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', or 'Sunday'"
+            + "or its 3-letter shorthands, example 'Mon' (non case-sensitive).";
+
+
+    public final DayOfWeek day;
+
+    /**
+     * Accepts String representations of a day of the calendar week. <br>
+     * Users can provide the day of the week in full, e.g. "Monday", "Saturday". <br>
+     * Experienced users may use the respective 3-letter shorthands to reduce typing required, e.g. "Tue" or "Fri". <br>
+     * In both cases, the check is non-case sensitive.
+     * @param day String representation of any day in the calendar week, e.g. "Monday" or "Wed".
+     */
+    public Day(String day) {
+        requireNonNull(day);
+        checkArgument(checkValidDayOfWeek(day), INVALID_DAY_OF_WEEK);
+
+        this.day = processDayOfWeek(day);
+    }
+
+    // Validator and parsing fns ================================================================================
+
+    /**
+     * Checks if the day of week matches any of the 7 days of the week of their 3-letter shorthands,
+     * example "Wednesday" or "fri".
+     * Non-case sensitive.
+     */
+    public static boolean checkValidDayOfWeek(String dayOfWeek) {
+        return DAYS_OF_THE_WEEK.contains(dayOfWeek.toLowerCase())
+            || DAYS_OF_THE_WEEK_SHORTHAND.contains(dayOfWeek.toLowerCase());
+    }
+
+    /**
+     * Generates a DayOfWeek based on a String representation.
+     * Callers are advised to use {@link #checkValidDayOfWeek(String)} first and handle abnormal use cases themselves.
+     * Otherwise, an error message is generated that may be visible to the user.
+     */
+    private static DayOfWeek processDayOfWeek(String dayOfWeek) {
+        checkArgument(checkValidDayOfWeek(dayOfWeek), INVALID_DAY_OF_WEEK);
+        switch (dayOfWeek.toLowerCase()) {
+        case "monday": case "mon":
+            return DayOfWeek.MONDAY;
+        case "tuesday": case "tue":
+            return DayOfWeek.TUESDAY;
+        case "wednesday": case "wed":
+            return DayOfWeek.WEDNESDAY;
+        case "thursday": case "thu":
+            return DayOfWeek.THURSDAY;
+        case "friday": case "fri":
+            return DayOfWeek.FRIDAY;
+        case "saturday": case "sat":
+            return DayOfWeek.SATURDAY;
+        case "sunday": case "sun":
+            return DayOfWeek.SUNDAY;
+        default:
+            // because this is prior to a validity check, this use of an unchecked runtime exception is
+            // warranted here as it denotes a gross failure of internal data processing.
+            throw new RuntimeException("Something went horribly wrong when parsing a lesson time.");
```

Comment:
I like how your switch case syntax follows the module style requirements

Reply:
thanks for the review!